src/pylibsshext/sftp.pyx

```diff
@@ -100,7 +100,7 @@ cdef class SFTP:
                 raise LibsshSFTPException("Reading data from remote file [%s] failed with error [%s]"
                                           % (remote_file, self._get_sftp_error_str()))
 
-            with open(local_file, 'wb+') as f:
+            with open(local_file, 'ab') as f:
```

Comment:
This will append content to the file which already exists. I've tested this diff with `ansible-libssh v1.2.2` and only a60a6cbf0ea4a7ef0e25299a1b3d01509488c6b5 applied. I see multiple executions on the same destination file growing above original size:

```
remote-machine# ls -l /tmp/foo?
-rw-r--r--  1 root  wheel  1048576 Nov 15 12:31 /tmp/foo1
-rwx------  1 root  wheel  3145728 Nov 15 12:37 /tmp/foo3
```

```
local-machine$ ls -l /tmp/foo?
-rw-r--r--  1 mkucharski  wheel  3145728 Nov 15 12:35 /tmp/foo2
```

Above we see size 3x after 3 executions of:
```
t1 = sftp_session.get(remote_file, local_file)
t2 = sftp_session.put(local_file, new_remote_file)
```

I think this needs more work. Also mode doesn't match after `put()`, but that is another issue.

Reply:
Thank you for throughful testing! Indeed, you are right. I thought this small change will make it working, but it looks like its more complicated. Updated the PR.

For the modes, I think this would be a different issue or rfe. Right now, the code opens the file with normal python open, which will probably set modes based on your umask, which will result in sane permissions. We could update the permissions based on what is on the server (if the server uses unix/linux ACL), but given that they will likely have different usernames and groups, it might result in unwanted permissions.